I added a pkg update to the Vagrantfile and that works.

However it still fails I think I have to pkg install -R into the new BE or reboot

 .... lots of messages by "vagrant up"

    default:  Actions: 22962/22969 actions (Updating modified actions)
    default:  Actions: Completed 22969 actions in 24.43 seconds.
    default:  Done
    default:  Done
    default:  Done
    default:  Done
    default: 
    default: A clone of openindiana-2021:03:02 exists and has been updated and activated.
    default: On the next boot the Boot Environment openindiana-2021:09:15 will be
    default: mounted on '/'.  Reboot when ready to switch to this updated BE.
    default: 
    default: 
    default: ---------------------------------------------------------------------------
    default: NOTE: Please review release notes posted at:
    default: 
    default: https://docs.openindiana.org/release-notes/latest-changes/
    default: ---------------------------------------------------------------------------
    default: 
    default:  Startup: Refreshing catalog 'openindiana.org' ... Done
    default: 
    default: pkg install: No matching version of metapackages/build-essential can be installed:
    default:   Reject:  pkg://openindiana.org/metapackages/build-essential@1.0-2020.0.1.22
    default:   Reason:  No version matching 'require' dependency crypto/gnupg can be installed
    default:     ----------------------------------------
    default:     Reject:  pkg://openindiana.org/crypto/gnupg@2.3.1-2020.0.1.0
    default:              pkg://openindiana.org/crypto/gnupg@2.3.2-2020.0.1.0
    default:     Reason:  This version is excluded by installed incorporation consolidation/userland/userland-incorporation@0.5.11-2020.0.1.13917
    default:     ----------------------------------------
    default: /bin/sh: git: not found [No such file or directory]
    default: make-rules/shared-macros.mk:1423: /make-rules/component.mk: No such file or directory
    default: gmake: *** No rule to make target '/make-rules/component.mk'.  Stop.
    default: Planning: Solver setup ... Startup: Refreshing catalog 'userland' ... Done
    default:  Startup: Caching catalogs ... Done
    default: VM is ready, happy contributing!

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -13,6 +13,7 @@
 #
 # Copyright 2016 Adam Stevko. All rights reserved.
 # Copyright 2017 Michal Nowak
+# Copyright 2021 David Stes
 #
 
 Vagrant.configure("2") do |config|
@@ -39,7 +40,10 @@ Vagrant.configure("2") do |config|
     host = RbConfig::CONFIG['host_os']
 
     # Get memory size and CPU cores amount
-    if host =~ /darwin/
+    if host =~ /solaris/
+	mem = `/usr/sbin/prtconf|grep Memory|cut -f3 -d' '`.to_i * 1024 * 1024
+	cpus = `/usr/sbin/psrinfo|wc -l`.to_i
+    elsif host =~ /darwin/
         # sysctl returns Bytes
         mem = `sysctl -n hw.memsize`.to_i
         cpus = `sysctl -n hw.ncpu`.to_i
@@ -51,6 +55,9 @@ Vagrant.configure("2") do |config|
         # Windows code via https://github.com/rdsubhas/vagrant-faster
         mem = `wmic computersystem Get TotalPhysicalMemory`.split[1].to_i
         cpus = `echo %NUMBER_OF_PROCESSORS%`.to_i
+    else
+	puts "Unsupported operating system"
+	exit
     end
 
     # Give VM 1/4 system memory as well as CPU core count
@@ -60,6 +67,7 @@ Vagrant.configure("2") do |config|
     config.vm.provider "virtualbox" do |v|
         v.customize ["modifyvm", :id, "--memory", mem]
         v.customize ["modifyvm", :id, "--cpus", cpus]
+
         v.customize ["storagectl", :id, "--name", "SATA Controller", "--hostiocache", "on"]
         # Enable following line, if oi-userland directory is on non-rotational
         # drive (e.g. SSD). (This could be automated, but with all those storage

